Automattic/jetpack
# Jetpack AI: cache the AI assistant feature data for Jetpack sites

@tomjn reported an issue where the app executes the exact requests multiple times in a brief interval. 
To prevent excessive requests to the dotcom side, this PR temporarily caches the first request for ten seconds, which reduces the initial rendering process time.

Fixes https://github.com/Automattic/jetpack/issues/33248

## Proposed changes:

* Jetpack AI: cache the AI assistant feature data for Jetpack sites

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* To test the cache system, you must add some logs in the code.

* Confirm the AI Assistant feature is propagated to the initial state.
  * Open the dev console of your dev client
  * Print the value of the following object

```js
Jetpack_Editor_Initial_State['ai-assistant']
```

* Confirm you see the data defined:
<img width="345" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/337ce712-6e08-4a32-bee9-6df5f3ca596b">

* Confirm the `ai-assistant-feature` endpoint response:
  * Open the Network tab of your dev client
  * Filter the requests by `ai`
  * Force the client to perform a request, closing and open the AI Jetpack panel, for instance
  * Confirm you see the response body

<img width="815" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/8d1abefe-57d8-425c-a27b-2b87403a18c1">
